gnovm/cmd/gno/clean.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/gnolang/gno/gnovm/pkg/gnomod"
+	"github.com/gnolang/gno/tm2/pkg/commands"
+	"github.com/gnolang/gno/tm2/pkg/crypto/keys/client"
+)
+
+type cleanCfg struct {
+	dryRun   bool // clean -n flag
+	verbose  bool // clean -x flag
+	modCache bool // clean -modcache flag
+}
+
+func newCleanCmd(io *commands.IO) *commands.Command {
+	cfg := &cleanCfg{}
+
+	return commands.NewCommand(
+		commands.Metadata{
+			Name:       "clean",
+			ShortUsage: "clean [flags]",
+			ShortHelp:  "Removes generated files and cached data",
+		},
+		cfg,
+		func(ctx context.Context, args []string) error {
+			return execClean(cfg, args, io)
+		},
+	)
+}
+
+func (c *cleanCfg) RegisterFlags(fs *flag.FlagSet) {
+	fs.BoolVar(
+		&c.dryRun,
+		"n",
+		false,
+		"print remove commands it would execute, but not run them",
+	)
+
+	fs.BoolVar(
+		&c.verbose,
+		"x",
+		false,
+		"print remove commands as it executes them",
+	)
+
+	fs.BoolVar(
+		&c.modCache,
+		"modcache",
+		false,
+		"remove the entire module download cache",
+	)
+}
+
+func execClean(cfg *cleanCfg, args []string, io *commands.IO) error {
+	if len(args) > 0 {
+		return flag.ErrHelp
+	}
+
+	path, err := os.Getwd()
+	if err != nil {
+		return err
+	}
+	modDir, err := gnomod.FindRootDir(path)
+	if err != nil {
+		return fmt.Errorf("not a gno module: %w", err)
+	}
+
+	if path != modDir && (cfg.dryRun || cfg.verbose) {
+		io.Println("cd", modDir)
+	}
+	err = filepath.WalkDir(modDir, func(path string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if d.IsDir() {
+			return nil
+		}
+		if !strings.HasSuffix(path, ".gno.gen.go") && !strings.HasSuffix(path, ".gno.gen_test.go") {
+			return nil
+		}
+		if !cfg.dryRun {
+			if err := os.Remove(path); err != nil {
+				return err
+			}
+		}
+		if cfg.dryRun || cfg.verbose {
+			io.Println("rm", strings.TrimPrefix(path, modDir+"/"))
+		}
+
```

Comment:
I think you can simplify these blocks and reduce this overhead of having a dry run / verbose output logic clutter main code:

You can keep track of the commands (ex. rm or rm -rf) and the dirs (strings) you are working with in the command in some type of structure array, and at the end of the command execution, you can simply iterate and print if the cry run or verbose flag is enabled

I think this method (`execClean`) has a lot of mental overhead when reading because you constantly do logic checking based on flags, and I believe we should go in the direction of simplifying it

Reply:
Current logic looks intuitive to me. 

Are you suggesting to append in the struct array instead of printing it right away? 
What if it errors in between? we need to print the list of removed dirs before exiting?